morse_decoder.rb

```diff
@@ -0,0 +1,54 @@
+MY_ABC = {
+  'a' => '.-',
+  'b' => '-...',
+  'c' => '-.-.',
+  'd' => '-..',
+  'e' => '.',
+  'f' => '..-.',
+  'g' => '--.',
+  'h' => '....',
+  'i' => '..',
+  'j' => '.---',
+  'k' => '-.-',
+  'l' => '.-..',
+  'm' => '--',
+  'n' => '-.',
+  'o' => '---',
+  'p' => '.--.',
+  'q' => '--.-',
+  'r' => '.-.',
+  's' => '...',
+  't' => '-',
+  'u' => '..-',
+  'v' => '...-',
+  'w' => '.--',
+  'x' => '-..-',
+  'y' => '-.--',
+  'z' => '--..'
+}.freeze
+
+def decode_char(char)
+  MY_ABC.each do |key, value|
+    return key.upcase if value == char
+  end
+end
+
+def decode_word(word)
+  new_word = ''
+  new_arr = word.split
+  new_arr.each do |value|
+    new_word += decode_char(value)
+  end
+  new_word
+end
+
+def decode(text)
+  new_word = ''
+  new_arr = text.split('   ')
+  new_arr.each do |value|
+    new_word += "#{decode_word(value)} "
+  end
+  puts new_word
+end
```

Comment:
- [ ] Hi team, nice job so far. Your `decode_char` and `decode_word` methods work as expected however, your decode method does not return the requested result, that's `"MY NAME"`. It rather returns `MY NAME `, next `nil` as seen on the screenshot below: 
![Screenshot from 2022-03-30 23-01-40](https://user-images.githubusercontent.com/63087491/160938385-551e04e8-a1f9-40a5-bc57-2a0cea50a916.png)

Please let's fix this issue.

- [ ] Also, let's make sure that the final result should be exactly the same as:
 
```
"A"
"MY"
"MY NAME"
"A BOX FULL OF RUBIES"
```
And **NOT**

"A"
"MY"
MY NAME 
nil
MY NAME 
A BOX FULL OF RUBIES 
nil
A BOX FULL OF RUBIES  

Thanks :pray:

Reply:
Hi, I tried a lot of times with different outputs but I never got the 'nil', do you know what could be the problem?

Reply:
![morse_decode](https://user-images.githubusercontent.com/73354132/161080470-eef49b35-af4f-497c-82a3-6cc5320d307f.PNG)

Hi @FrederickMih, this is the output that we get from our project.

Reply:
![Screenshot from 2022-03-30 21-20-18](https://user-images.githubusercontent.com/64876126/161084669-cb1341e1-00e4-47cc-8cc9-422db0193539.png)

As you can see, is also the same on my computer, kindly answer so we could get this cleared

Reply:
Hi, yeah, that's for the IRB, which is always returning nil when we run a "puts", but in our Vscode it was working well so we weren't able to realize how to solve the problem, thanks for taking the time to answer anyway, I appreciate it!